src/util/address.rs

```diff
@@ -397,11 +401,11 @@ impl Payload {
             Payload::ScriptHash(ScriptHash::from_inner(hash_inner))
         } else if script.is_witness_program() {
             Payload::WitnessProgram {
```

Comment:
From https://github.com/rust-bitcoin/rust-bitcoin/issues/1022#issuecomment-1144390900, would it benefit to return an error indicating incorrect segwit v0 program length here?

Then we can remove the explicit check in `Address::from_script()` since it would be done here.

Reply:
Good point. That way, we can reduce the number of calls to `script.is_witness_program()` for correct scripts.